applications/GeoMechanicsApplication/custom_strategies/strategies/README.md

```diff
@@ -0,0 +1,93 @@
+# GeoMechanicsNewtonRaphsonErosionProcessStrategy class.
+
+This class implements a Newton-Raphson process for the backward erosion piping. The physics of the piping process is
+described in
+
+1. van Beek M. Backward erosion piping,initiation and progression. TUDelft PhD. Thesis, 2015.
+2. Technisch rapport Zandmeevoerende wellen. Technische Adviescommissie
+   voor de Waterkeringen, Delft, The Netherlands, 1999.
+
+## Algorithm
+
+The algorithm is implemented in FinalizeSolutionStep function that loops over all piping elements until the erosion
+growth is finished or all piping elements are checked.
+
+### FinalizeSolutionStep function
+
+![FinalizeSolutionStep.svg](FinalizeSolutionStep.svg)
+
+This function calls other functions that are described below.
+
+### GetPipingElements function
+
+This function returns a list of all piping elements. To create the list, all elements are checked for  
+PIPE_START_ELEMENT property that holds Id of the piping starting element.
+Then elements on the list are sorted based on X-coordinate of the element begin.
+The current limitation is the following. The starting piping element has to have minimum or maximum value of the X
+coordinate. This is needed to find a direction for the piping growth.
+
+### InitialiseNumActivePipeElements function
+
+It checks the list of the elements and returns a number of elements with PIPE_ACTIVE property At the beginning the
+number is zero.
+
+### CalculateMaxPipeHeight function
+
+It calculates a maximum diameter using CalculateParticleDiameter function then multiplies it by a magic factor of 100 to
+get the height value.
+
+### CalculateParticleDiameter
+
+In case of PIPE_MODIFIED_D, the function returns
+
+$$ diameter = 2.08\times 10^{-4} * \Bigg(\frac{D_{70}}{2.08\times 10^{-4}}\Bigg)^{0.4}$$
+
+otherwise, $$diameter = D_{70}$$
+
+### check_pipe_equilibrium function
+
+This function performs iterative process with a prescribed maximum number of iterations. For each iteration the
+modelling is
+performed. Then a pipe height is calculated for each open piping element.
```

Comment:
I think it would be beneficial to emphasize that the reason this function iterates is to find the correct pipe height for the elements. Since the function name is not completely clear, I think it's nice to say that is the end goal (if I understand correctly)

Reply:
You are right, the description is changed.